### PR TITLE
perf: optimize `Misc.compute_committee` a bit

### PR DESCRIPTION
**Pull request by MegaRedHand**
This reduces the time of `process_attestation` by ~0.5s (~30%)

### CODE REVIEW

File: lib/lambda_ethereum_consensus/state_transition/misc.ex [hunk 1]
```diff
@@ -182,25 +182,29 @@ defmodule LambdaEthereumConsensus.StateTransition.Misc do
...
-    Enum.reduce_while(start_..end_, {:ok, []}, fn i, {:ok, acc_list} ->
-      case compute_shuffled_index(i, length(indices), seed) do
-        {:ok, shuffled_index} ->
-          {:cont, {:ok, [Enum.at(indices, shuffled_index) | acc_list]}}
-
-        {:error, _} = error ->
-          {:halt, error}
-      end
-    end)
+    index_count = length(indices)
```
Comment (Arkenan) [inline lib/lambda_ethereum_consensus/state_transition/misc.ex:185]: Nit: can we add a comment in the function as to what each parameter is: e.g. indices is the list of validator indices, index is the index of the committee being computed, count is the number of committees in the epoch.

File: lib/lambda_ethereum_consensus/state_transition/misc.ex [hunk 2]
```diff
@@ -182,25 +182,29 @@ defmodule LambdaEthereumConsensus.StateTransition.Misc do
...
+    a = div(index_count * index, count)
+    b = div(index_count * (index + 1), count) - 1
```
Comment (Arkenan) [inline lib/lambda_ethereum_consensus/state_transition/misc.ex:187]: Nit: can we change a and b for `committee_start` and `committee_end`?

**Review by mpaulucci — APPROVED** [review]

File: lib/lambda_ethereum_consensus/state_transition/misc.ex [hunk 3]
```diff
@@ -182,25 +182,29 @@ defmodule LambdaEthereumConsensus.StateTransition.Misc do
...
+
+    to_swap_indices =
+      a..b//1
+      |> Stream.map(&compute_shuffled_index(&1, index_count, seed))
+      |> Stream.map(fn {:ok, i} -> i end)
```
Comment (Arkenan) [inline lib/lambda_ethereum_consensus/state_transition/misc.ex:192]: What if this is not `:ok`? We should fix that edge case. E.g. with a reduce similar to what we use in SSZ when we have to encode several types and they fail.

**Review by Arkenan — APPROVED** [review]
This PR looks great! We only need to take the case where compute_shuffled_index fails into account.

File: lib/lambda_ethereum_consensus/state_transition/misc.ex (hunk 3 above)
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/state_transition/misc.ex]: Done

**Pull request merged**